hazelcast/src/main/java/com/hazelcast/config/CacheConfig.java

```diff
@@ -519,6 +543,10 @@ public void writeData(ObjectDataOutput out) throws IOException {
         out.writeBoolean(disablePerEntryInvalidationEvents);
 
         writePartitionLostListenerConfigs(out);
+
+        if (out.getVersion().isGreaterOrEqual(Versions.V5_0)) {
```

Comment:
nit: mark this and other rolling-upgrade related changes with a `// RU_COMPAT_4_2` comment so they can be easily spotted

Reply:
done